datafusion/common/src/config.rs

```diff
@@ -1555,7 +1555,7 @@ config_namespace! {
         /// Specifies whether there is a CSV header (i.e. the first line
         /// consists of is column names). The value `None` indicates that
         /// the configuration should be consulted.
-        pub has_header: Option<bool>, default = None
+      pub has_header: bool, default = false
```

Comment:
I don't understand the implication of this change and there are no tests that cover it.

Can you please at least update the documentation to reflect the new behavior (it can no longer be None)

Also there appears to be some strange indenting going on